Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/MainActivity.kt

```diff
@@ -167,7 +166,9 @@ class MainActivity : AppCompatActivity(), HasAndroidInjector {
                 is MainActivityEvent.GoToCheckInsFragment -> navController.navigate(
                     CheckInsFragment.createDeepLink(event.uriString)
                 )
-                is MainActivityEvent.Error -> event.error.toErrorDialogBuilder(this).show()
+                is MainActivityEvent.Error -> displayDialog {
+                    event.error.toErrorDialogBuilder(context)
```

Comment:
this should be passed as `dialog=event.error.toErrorDialogBuilder(context)` otherwise it does not have an effect , right?

Reply:
Ah yes, you're right. I'll update it.

Reply:
Not sure about the refactoring. We'd go back to a constructor with many params. I like the lambda way of configuring the dialog. We just need to make sure to not put anything else in here, same as with any lambda.